aptos-move/block-executor/src/view.rs

```diff
@@ -472,10 +484,48 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                 }
             },
             ViewState::Unsync(state) => {
-                let ret = state.unsync_map.fetch_data(state_key).map_or_else(
-                    || self.get_base_value(state_key),
-                    |v| Ok(v.as_state_value()),
-                );
+                let ret = match state.unsync_map.fetch_data(state_key) {
+                    // TODO: Should we ignore the layout from fetch_data?
+                    // What if it doesn't match with the maybe_layout given as input to this function?
+                    Some((v, _)) => Ok(v.as_state_value()),
+                    None => {
+                        self.get_base_value(state_key).map(
+                            |maybe_state_value| {
+                                if let ReadKind::Value = kind {
+                                    match (maybe_state_value, maybe_layout) {
```

Comment:
I think using `match (kind, maybe_state_value, maybe_layout)` will help to get rid of `if let`?

Reply:
Done/